logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LoginUseCase.kt

```diff
@@ -133,16 +136,23 @@ internal class LoginUseCaseImpl internal constructor(
             .fold({
                 when (it) {
                     is NetworkFailure.ProxyError -> AuthenticationResult.Failure.SocketError
-                    is NetworkFailure.ServerMiscommunication -> handleServerMiscommunication(it)
+                    is NetworkFailure.ServerMiscommunication -> handleServerMiscommunication(it, isEmail, cleanUserIdentifier)
                     is NetworkFailure.NoNetworkConnection, NetworkFailure.FederatedBackendFailure ->
                         AuthenticationResult.Failure.Generic(it)
                 }
             }, {
+                if (isEmail && secondFactorVerificationCode != null) {
+                    secondFactorVerificationRepository.storeVerificationCode(cleanUserIdentifier, secondFactorVerificationCode)
+                }
                 it
             })
     }
 
-    private fun handleServerMiscommunication(error: NetworkFailure.ServerMiscommunication): AuthenticationResult.Failure {
+    private suspend fun handleServerMiscommunication(
+        error: NetworkFailure.ServerMiscommunication,
+        isEmail: Boolean,
+        userIdentifier: String,
```

Comment:
this is also abit weird and unclear to me what is happening, maybe perhaps we can leave the function as it was and call this somewhere in the place where we are handelingServerMiscommuication, beecause before it was simply mapping stuff, but now it has a side effect, which I am not directly aware of.